### PR TITLE
Miawong/sc 67855/refactor dashboard fetch into hooks 

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it:
- adds hooks for 5 of the fetches that occur in the Dashboard.tsx (fetchAppDownstream, updateStatus, createSnapshot, getAirgapConfig, onChangeForUpdates) 
- naming and types may be updated in subsequent PRs as we integrate these hooks 

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/67855/refactor-dashboard-api-call-to-use-types-selector-and-refetchinterval-react-query

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/features/Dashboard/api/createSnapshot.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+const createSnapshotResponse = (response: SnapshotResponse): Snapshot => {
+  return {
+    startingSnapshot: response.kotsadmRequiresVeleroAccess ? false : true,
+  };
+};
+
+export const useCreateSnapshot = (option: string) => {
+  const { selectedApp } = useSelectedApp();
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/createSnapshot.ts:47]: glad to see this being used. wish i could use it in preflights stuff

File: web/src/features/Dashboard/api/createSnapshot.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  return response;
+};
+
+const createSnapshotResponse = (response: SnapshotResponse): Snapshot => {
+  return {
+    startingSnapshot: response.kotsadmRequiresVeleroAccess ? false : true,
+  };
+};
+
+export const useCreateSnapshot = (option: string) => {
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/createSnapshot.ts:46]:
```suggestion
export const useCreateSnapshot = (option?: "full") => {
```
could constrain the type with something like this

File: web/src/features/Dashboard/api/getAirgapConfig.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  let res = await fetch(configUrl, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: Utilities.getToken(),
+    },
+  });
+  const response = await res.json();
+  if (res.ok) {
+    simultaneousUploads = response.simultaneousUploads;
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/getAirgapConfig.ts:19]: oh i should start doing this more

File: web/src/features/Dashboard/api/getUpdateDownloadStatus.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+export const useUpdateDownloadStatus = () => {
+  const { selectedApp } = useSelectedApp();
+
+  return useQuery({
+    queryFn: () => getUpdateDownloadStatus(selectedApp?.slug || ""),
+    queryKey: ["getUpdateStatus"],
+    onError: (err: Error) => console.log(err),
+    refetchInterval: (data) => (data?.status !== "running" ? false : 1000),
+    select: (response: UpdateStatusResponse) =>
+      makeUpdateStatusResponse(response),
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/getUpdateDownloadStatus.ts:57]:
```suggestion
    select: makeUpdateStatusResponse,
```

I think this works too unless you're planning on having some state in this hook to pass in.

File: web/src/features/Dashboard/api/getUpdateDownloadStatus.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  );
+
+  if (!res.ok) {
+    throw new Error("Error getting update status");
+  }
+  const appResponse = await res.json();
+  return appResponse;
+};
+
+const makeUpdateStatusResponse = (
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/getUpdateDownloadStatus.ts:37]: nice

**Review by rianfowler — DISMISSED** [review]
looks good!

File: web/src/features/Dashboard/api/createSnapshot.ts (hunk 1 above)
Comment (miaawong) [reply web/src/features/Dashboard/api/createSnapshot.ts:47]: it's amazing 🙌

**Review by rianfowler — APPROVED** [review]

**Pull request merged**